### PR TITLE
recieve handlers should never send as they may block forever if the u…

### PR DESCRIPTION
**Pull request by plorenz**
…nderlying channel closes

### CODE REVIEW

File: controller/handler_ctrl/create_terminator.go [hunk 1]
```diff
@@ -46,10 +47,13 @@ func (h *createTerminatorHandler) ContentType() int32 {
 func (h *createTerminatorHandler) HandleReceive(msg *channel2.Message, ch channel2.Channel) {
 	request := &ctrl_pb.CreateTerminatorRequest{}
 	if err := proto.Unmarshal(msg.Body, request); err != nil {
-		handler_common.SendFailure(msg, ch, err.Error())
+		log.WithError(err).Error("failed to unmarshal create terminator message")
```
Comment (andrewpmartinez) [inline controller/handler_ctrl/create_terminator.go:50]: `SendResult` used to do the following:

```
	log := pfxlog.ContextLogger(ch.Label())
	if !success {
		log.Errorf("%v error (%s)", ch.LogicalName(), message)
	}
```

This new log message has less information in it. Was that intentional?

File: controller/handler_ctrl/inspect.go [hunk 2]
```diff
@@ -42,10 +41,14 @@ func (h *inspectHandler) HandleReceive(request *channel2.Message, ch channel2.Ch
 
 	inspectRequest := &ctrl_pb.InspectRequest{}
 	if err := proto.Unmarshal(request.Body, inspectRequest); err != nil {
-		log.Errorf("unexpected error (%s)", err)
-		h.respondWithError(ch, request, fmt.Errorf("failed to decode request: (%v)", err).Error())
+		log.WithError(err).Error("failed to unmarshal inspect message")
```
Comment (andrewpmartinez) [inline controller/handler_ctrl/inspect.go:44]: I see that handleInspect does reply but this if block simply returns and doesn't send a response.

So there is no one that was waiting for the response?

File: controller/handler_ctrl/create_terminator.go [hunk 3]
```diff
@@ -24,6 +24,7 @@ import (
 	"github.com/openziti/fabric/controller/network"
 	"github.com/openziti/fabric/pb/ctrl_pb"
 	"github.com/openziti/foundation/channel2"
+	log "github.com/sirupsen/logrus"
```
[the remark this answers is not in the document]
Comment (plorenz) [reply controller/handler_ctrl/create_terminator.go]: I had copied the log message from elsewhere. This code didn't already have a `log := pfxlog.ContextLogger...` so Goland 'helpfully' assumed I meant logrus with log. Thanks for catching that, it's now fixed.

File: controller/handler_ctrl/create_terminator.go (hunk 1 above)
Comment (plorenz) [reply controller/handler_ctrl/create_terminator.go:50]: Now that I've added the `log ;= pfxlog...` bit back in, there's more info than before. The context logger has the channel label which includes the channel logical name. We're also logging the error and some extra info about what kind of message we were expecting.

**Review by andrewpmartinez — APPROVED** [review]

**Pull request merged**